File: src/Content/NetCoreTool.Template.WebApi/FSharp/HelloHystrixCommand.fs
namespace Company.WebApplication.FS
#if (CircuitBreakerHystrixOption)

open System.Threading.Tasks
open Steeltoe.CircuitBreaker.Hystrix

type HelloHystrixCommand(name : string) =
    inherit HystrixCommand(HystrixCommandGroupKeyDefault.AsKey(name))
#endif

File: src/Content/NetCoreTool.Template.WebApi/CSharp/HelloHystrixCommand.cs
#if (HasCircuitBreakerHystrixInSteeltoeV3)
using Steeltoe.CircuitBreaker.Hystrix;

namespace Company.WebApplication.CS
{
    public sealed class HelloHystrixCommand : HystrixCommand<string>
    {
        private readonly string _name;

        public HelloHystrixCommand(string name)
            : base(HystrixCommandGroupKeyDefault.AsKey("ExampleCircuitBreakers"))
        {
            _name = name;
            IsFallbackUserDefined = true;
        }

        protected override async Task<string> RunAsync()
        {
            return await Task.FromResult("Hello " + _name);
        }

        protected override async Task<string> RunFallbackAsync()
        {
            return await Task.FromResult("Hello " + _name + " via fallback");
        }
    }
}
#endif
